### PR TITLE
adjusted bagging_freq parameter description

### PR DESCRIPTION
**Pull request by moziada**

### CODE REVIEW

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for your interest in LightGBM and for taking the time to modify the documentation.

I'm ok with this change to the docs. However, we have a slightly different process for modifying the `Parameters` documentation page.

Please modify this:

https://github.com/microsoft/LightGBM/blob/4f47547c88cc72de4b8f04de8de4e2b79a9240e7/include/LightGBM/config.h#L336

Then run this from the root of the repo

```shell
python helpers/parameter_generator.py
```

and commit the resulting changes.

-------

In the future, I strongly recommend making pull requests from a new branch on your fork, instead of the target repo's default branch (`master`, in LightGBM's case). This project uses squash-merging, so `master` on your fork and `master` here in the main repository are going to have different histories now that you've created a pull request from `master` in your fork.

Once this pull request is merged, I recommend deleting your fork of LightGBM and then re-forking if you want to make further contributions here.

**Comment by moziada** [comment]
Hi James,
After editing ```config.h``` and running ```python helpers/parameter_generator.py``` command as you mentioned, git showed that ```config.h``` is only the file that has changes.
And sure for future contributions I will make a pull requests from a new branch rather than ```master```

**Comment by moziada** [comment]
> @moziada please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

agree

File: include/LightGBM/config.h [hunk 1]
```diff
@@ -333,7 +333,7 @@ struct Config {
   // alias = subsample_freq
   // desc = frequency for bagging
   // desc = ``0`` means disable bagging; ``k`` means perform bagging at every ``k`` iteration. Every ``k``-th iteration, LightGBM will randomly select ``bagging_fraction * 100 %`` of the data to use for the next ``k`` iterations
-  // desc = **Note**: to enable bagging, ``bagging_fraction`` should be set to value smaller than ``1.0`` as well
+  // desc = **Note**: to enable bagging, ``bagging_fraction`` should be set to a value between ``0.0`` and ``1.0``
```
Comment (jmoralez) [inline include/LightGBM/config.h:336]: Should we maybe change the wording a bit? The original intent is to clarify that `bagging_fraction=1.0` disables bagging. Between usually means inclusive on both sides, but this param should be greater than 0 and less than 1 to enable it.

Comment (moziada) [reply include/LightGBM/config.h]: > Should we maybe change the wording a bit? The original intent is to clarify that `bagging_fraction=1.0` disables bagging. Between usually means inclusive on both sides, but this param should be greater than 0 and less than 1 to enable it.

Clarification is made

**Comment by moziada** [comment]
@microsoft-github-policy-service agree

File: include/LightGBM/config.h (hunk 1 above)
Comment (moziada) [reply include/LightGBM/config.h]: sure

**Comment by jameslamb** [comment]
> After editing config.h and running python helpers/parameter_generator.py command as you mentioned, git showed that config.h is only the file that has changes

Thanks for doing that! That's because you'd already manually changed `docs/Parameters.rst`. If you had changed `config.h` then run that Python script, it would have altered `docs/Parameters.rst`.

**Review by jameslamb — APPROVED** [review]
works for me, thanks!

**Comment by jameslamb** [comment]
Looks like all of @jmoralez 's suggestions have been addressed, so I'm going to merge this.

Thanks for taking the time to contribute 👋🏻

**Pull request merged**